python/hail/dataset.py

```diff
@@ -5482,5 +5482,52 @@ def to_hail2(self):
         import hail2
         return hail2.VariantDataset(self.hc, self._jvds)
 
+    @handle_py4j
+    @record_method
+    @typecheck_method(pedigree=Pedigree,
+               complete_trios=bool)
+    def trio_matrix(self, pedigree, complete_trios=False):
+        """Builds and returns a matrix with trios are grouped together as cells.
+
+        **Examples**
+
+        Create a trio matrix:
+
+        >>> pedigree = Pedigree.read('data/myStudy.fam')
+        >>> trio_matrix = vds.trio_matrix(pedigree, complete_trios=True)
+
+        **Notes**
+
+        This method builds a new dataset with trio columns, with one column
+        per trio (with ``complete_trios == False``) or complete trio (with
+        ``complete_trios == True``). In this new dataset, the column identifiers
+        are the sample IDs of the trio probands. The column annotations and
+        cells of the matrix are changed in the following ways:
+
+        The new column annotation schema is a ``Struct`` with three ``Struct`` children,
+        each with an ``id`` and ``annotations`` field. The ``annotations`` fields have the
+        same schema as the column annotation schema of the input dataset.
+
+         - **proband.id** (*String*) - Proband sample ID.
+         - **proband.annotations** (*Struct*) - Annotations on the proband.
+         - **mother.id** (*String*) - Mother sample ID.
```

Comment:
In fact, comparing this:
```
    val newColumnAnnotationType = TStruct(
      "proband" -> memberAnnotationType,
      "mother" -> memberAnnotationType,
      "father" -> memberAnnotationType
    )
```
and this
```
newSampleAnnotations(i) = Row(kidAnnotation, dadAnnotation, momAnnotation)
```
this switch induced a bug that you're not catching because your test doesn't include sample annotations.